src/main/kotlin/com/dansplugins/factionsystem/dynmap/MfDynmapService.kt

```diff
@@ -51,265 +76,170 @@ class MfDynmapService(private val plugin: MedievalFactions) {
         }
     }
 
-    fun updateClaims(faction: MfFaction) {
+    private fun updateClaims(faction: MfFaction) {
+        updateClaimsInvocationCount++
+        if (debug) {
+            plugin.logger.info("Updating ${faction.name} claims. Invocation count: $updateClaimsInvocationCount")
+        }
         val markerApi = dynmap.markerAPI
         if (markerApi == null) {
             plugin.logger.warning("Failed to find Dynmap Marker API, skipping update of ${faction.name} claims")
             return
         }
-        val claimsMarkerSet = markerApi.getMarkerSet("claims")
-            ?: markerApi.createMarkerSet("claims", "Claims", null, false)
-        val realmMarkerSet = markerApi.getMarkerSet("realms")
-            ?: markerApi.createMarkerSet("realms", "Realms", null, false)
+        val claimsMarkerSet = markerSetHelper.getOrCreateMarkerSet(markerApi, "claims", "Claims")
+        val realmMarkerSet = markerSetHelper.getOrCreateMarkerSet(markerApi, "realms", "Realms")
```

Comment:
Passing the markerApi object into the MarkerSetHelper when it is constructed would simplify these calls slightly

Reply:
Ignoring